contracts/modules/cover/Cover.sol

```diff
@@ -92,22 +96,17 @@ contract Cover is ICover, MasterAwareV2, IStakingPoolBeacon {
   constructor(
     IQuotationData _quotationData,
     IProductsV1 _productsV1,
-    address _stakingPoolImplementation,
     address _coverNFT,
-    address coverProxy
+    address _stakingPoolImplementation
   ) {
 
     // initialize immutable fields only
     quotationData = _quotationData;
     productsV1 = _productsV1;
-    stakingPoolProxyCodeHash = keccak256(
-      abi.encodePacked(
-        type(MinimalBeaconProxy).creationCode,
-        abi.encode(coverProxy)
-      )
-    );
-    stakingPoolImplementation =  _stakingPoolImplementation;
     coverNFT = _coverNFT;
+
+    stakingPoolProxyCodeHash = CoverUtilsLib.calculateProxyCodeHash();
+    stakingPoolImplementation = _stakingPoolImplementation;
```

Comment:
Use `_stakingPoolImplementation.codehash` instead of `CoverUtilsLib.calculateProxyCodeHash()`

Reply:
LE: not applicable here.